ambari-logsearch-web/src/app/modules/shared/components/user-settings/user-settings.component.ts

```diff
@@ -0,0 +1,123 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Component, Input, OnInit, OnDestroy } from '@angular/core';
+import { FormGroup, FormControl } from '@angular/forms';
+
+import { Subject } from 'rxjs/Subject';
+import { Observable } from 'rxjs/Observable';
+import { Router, ActivatedRoute } from '@angular/router';
+
+import { Store } from '@ngrx/store';
+import { AppStore } from '@app/classes/models/store';
+import { selectDisplayShortHostNames, selectTimeZone } from '@app/store/selectors/user-settings.selectors';
+import { selectHostNames } from '@app/store/selectors/hosts.selectors';
+import { SetUserSettingsAction } from '@app/store/actions/user-settings.actions';
+
+@Component({
+  selector: 'user-settings',
+  templateUrl: './user-settings.component.html',
+  styleUrls: ['./user-settings.component.less']
+})
+export class UserSettingsComponent implements OnInit, OnDestroy {
+
+  @Input()
+  visibilityQueryParamName = 'showUserSettings';
+
+  demoHostName = 'subdomain01.company-domain.exp';
+  
+  visible$: Observable<boolean> = this.route.queryParams
```

Comment:
Some methods here are the same as in time zone picker component. Wouldn't it be better to move them to some shared class/interface?

Reply:
Good catch. I had discussion about doing a common way to open modals. At the I decided to defer this task in order to
- not to delay the shipping this feature 
- have time to have research around how can it be optimum way

I opened a ticket for this: AMBARI-25009 to be sure that it won't become forgotten.